src/assets.rs

```diff
@@ -0,0 +1,364 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Seahorse library.
+
+// This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+// You should have received a copy of the GNU General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! The assets module.
+//!
+//! This module defines [Asset], [AssetEditor], and [Assets], which provide CURD (create, read,
+//! update, and delete) operations, with the use of [KeyValueStore] to control the assets resource.
+
+use crate::{asset_library::Icon, key_value_store::*, KeystoreError, Ledger, MintInfo};
+use jf_cap::structs::{AssetCode, AssetDefinition, AssetPolicy};
+use serde::{Deserialize, Serialize};
+use std::ops::{Deref, DerefMut};
+
+/// An asset with its code as the primary key.
+#[derive(Clone, Deserialize, Serialize)]
+pub struct Asset {
+    definition: AssetDefinition,
+    /// Optional asset name.
+    pub name: Option<String>,
+    /// Optional asset description.
+    pub description: Option<String>,
+    /// Optional asset icon.
+    pub icon: Option<Icon>,
+    mint_info: Option<MintInfo>,
+    verified: bool,
+    temporary: bool,
+}
+
+impl Asset {
+    #![allow(dead_code)]
+
+    /// Create a verified and temporary asset without mint information.
+    ///
+    /// Returns the created asset.
+    fn new_verified(definition: AssetDefinition) -> Self {
```

Comment:
I think this is more convenient as a way to create a verified asset from an existing `Asset`, a la `AssetInfo::verified` in the old implementation. Something like

```rust
fn export_verified(mut self) -> Self {
  self.mint_info = None;
  self.verified = true;
}
```